### ISSUE TITLE
Rollup warnings

### ISSUE THREAD

**Issue by vovsemenv** [adobe/react-spectrum#6933]
### Provide a general summary of the issue here

when i build with `vite@5.4.0` i get this warnings

```
"/*#__PURE__*/"

in "node_modules/react-aria-components/dist/Breadcrumbs.mjs" contains an annotation that Rollup cannot interpret due to the position of the comment. The comment will be removed to avoid issues.
node_modules/react-aria-components/dist/ComboBox.mjs (49:42): A comment

"/*#__PURE__*/"

in "node_modules/react-aria-components/dist/ComboBox.mjs" contains an annotation that Rollup cannot interpret due to the position of the comment. The comment will be removed to avoid issues.
```

### 🤔 Expected Behavior?

no warnings

### 😯 Current Behavior

warnings

### 🖥️ Steps to Reproduce

-

### Version

1.3.1

### What browsers are you seeing the problem on?

Other

### What operating system are you using?

macos

**Comment by snowystinger** [comment]
Sounds like a rollup or vite issue? have you checked their repos for that error or any bugs around it?

Otherwise, we'd need to know more about why it's thinking that it's invalid https://rollupjs.org/configuration-options/#pure but nothing is jumping out at me, other than I'm not actually sure what is generating the annotation. That's just a provider from a createContext call, so possibly React is generating it. Could check React and babel's docs to see if they have any bugs as well.

For reference, this the file and the Pure annotation is on line 49 https://www.unpkg.com/browse/react-aria-components@1.3.3/dist/ComboBox.mjs

**Comment by synecdokey** [comment]
Something jumping out to me is the fact that examples show the comment placed *before* a statement, not in the middle of it like the ComboBox generated code.

**Comment by hosenur** [comment]
Any fix for this ?

**Comment by snowystinger** [comment]
Not yet, someone will need to figure out if the PURE annotation is actually a bug in rollup or if it's a bug in our transpiled code first.

**Comment by matt-aaron** [comment]
The `createElement` function is what should be marked with the `/*#__PURE__*/` annotation. It appears the annotation is being misplaced due to the transform of the nullish coalescing operator.

I was able to recreate the issue in a simplified [Parcel REPL](https://repl.parceljs.org/#JTdCJTIyZmlsZXMlMjIlM0ElNUIlNUIlMjIlMkZzcmMlMkZpbmRleC5odG1sJTIyJTJDJTdCJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJ2YWx1ZSUyMiUzQSUyMiUzQyFET0NUWVBFJTIwaHRtbCUzRSU1Q24lM0NtYWluJTNFJTNDJTJGbWFpbiUzRSU1Q24lM0NzY3JpcHQlMjBzcmMlM0QlNUMlMjIuJTJGaW5kZXguanN4JTVDJTIyJTIwdHlwZSUzRCU1QyUyMm1vZHVsZSU1QyUyMiUzRSUzQyUyRnNjcmlwdCUzRSUyMiU3RCU1RCUyQyU1QiUyMiUyRnNyYyUyRmluZGV4LmpzeCUyMiUyQyU3QiUyMnZhbHVlJTIyJTNBJTIyaW1wb3J0JTIwKiUyMGFzJTIwUmVhY3QlMjBmcm9tJTIwJTVDJTIycmVhY3QlNUMlMjIlM0IlNUNuaW1wb3J0JTIwJTdCJTIwY3JlYXRlUm9vdCUyMCU3RCUyMGZyb20lMjAncmVhY3QtZG9tJTJGY2xpZW50JyUzQiU1Q25pbXBvcnQlMjAlN0JBcHAlN0QlMjBmcm9tJTIwJy4lMkZBcHAuanN4JyUzQiU1Q25sZXQlMjByb290JTIwJTNEJTIwY3JlYXRlUm9vdChkb2N1bWVudC5xdWVyeVNlbGVjdG9yKCU1QyUyMm1haW4lNUMlMjIpKSUzQiU1Q25yb290LnJlbmRlciglM0NBcHAlMjAlMkYlM0UpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlNUIlMjIlMkZzcmMlMkZBcHAuanN4JTIyJTJDJTdCJTIydmFsdWUlMjIlM0ElMjJpbXBvcnQlMjAqJTIwYXMlMjBSZWFjdCUyMGZyb20lMjAlNUMlMjJyZWFjdCU1QyUyMiUzQiU1Q24lNUNuY29uc3QlMjBNeUNvbnRleHQlMjAlM0QlMjBSZWFjdC5jcmVhdGVDb250ZXh0KG51bGwpJTNCJTVDbiU1Q25mdW5jdGlvbiUyMENvbWJvQm94KHByb3BzKSUyMCU3QiU1Q24lMjAlMjBsZXQlMjBjb250ZW50JTIwJTNEJTIwdXNlTWVtbygoKSUyMCUzRCUzRSUyMCglNUNuJTIwJTIwJTIwJTIwJTNDTXlDb250ZXh0LlByb3ZpZGVyJTIwdmFsdWUlM0QlN0IlN0IlMjBpdGVtcyUzQSUyMHByb3BzLml0ZW1zJTIwJTNGJTNGJTIwcHJvcHMuZGVmYXVsdEl0ZW1zJTIwJTdEJTdEJTNFJTVDbiUyMCUyMCUyMCUyMCUyMCUyMENvbnRlbnQlNUNuJTIwJTIwJTIwJTIwJTNDJTJGTXlDb250ZXh0LlByb3ZpZGVyJTNFJTVDbiUyMCUyMCklMkMlMjAlNUJwcm9wcy5pdGVtcyUyQyUyMHByb3BzLmRlZmF1bHRJdGVtcyU1RCklM0IlNUNuJTVDbiUyMCUyMHJldHVybiUyMCUzQ2RpdiUzRSU3QmNvbnRlbnQlN0QlM0MlMkZkaXYlM0UlM0IlNUNuJTdEJTVDbiU1Q25leHBvcnQlMjBmdW5jdGlvbiUyMEFwcCgpJTIwJTdCJTVDbiUyMCUyMHJldHVybiUyMCglNUNuJTIwJTIwJTIwJTIwJTNDQ29tYm9Cb3glMjBpdGVtcyUzRCU3QiU1QjElMkMlMjAyJTJDJTIwMyU1RCU3RCUyMGRlZmF1bHRJdGVtcyUzRCU3QiU1QjklMkMlMjA5JTJDJTIwOSU1RCU3RCUyMCUyRiUzRSU1Q24lMjAlMjApJTNCJTVDbiU3RCU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCU1RCUyQyU1QiUyMiUyRnBhY2thZ2UuanNvbiUyMiUyQyU3QiUyMnZhbHVlJTIyJTNBJTIyJTdCJTVDbiUyMCUyMCUyMCUyMCU1QyUyMm5hbWUlNUMlMjIlM0ElMjAlNUMlMjJyZXBsJTVDJTIyJTJDJTVDbiUyMCUyMCUyMCUyMCU1QyUyMnZlcnNpb24lNUMlMjIlM0ElMjAlNUMlMjIwLjAuMCU1QyUyMiUyQyU1Q24lMjAlMjAlMjAlMjAlNUMlMjJlbmdpbmVzJTVDJTIyJTNBJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU1QyUyMmJyb3dzZXJzJTVDJTIyJTNBJTIwJTVDJTIyc2luY2UlMjAyMDE5JTVDJTIyJTVDbiUyMCUyMCUyMCUyMCU3RCUyQyU1Q24lMjAlMjAlMjAlMjAlNUMlMjJ0YXJnZXRzJTVDJTIyJTNBJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU1QyUyMmFwcCU1QyUyMiUzQSUyMCU3QiU3RCU1Q24lMjAlMjAlMjAlMjAlN0QlMkMlNUNuJTIwJTIwJTIwJTIwJTVDJTIyZGVwZW5kZW5jaWVzJTVDJTIyJTNBJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU1QyUyMnJlYWN0JTVDJTIyJTNBJTIwJTVDJTIyKiU1QyUyMiUyQyU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlNUMlMjJyZWFjdC1kb20lNUMlMjIlM0ElMjAlNUMlMjIqJTVDJTIyJTVDbiUyMCUyMCUyMCUyMCU3RCU1Q24lN0QlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCU1RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJlbnRyaWVzJTIyJTNBJTVCJTVEJTJDJTIybWluaWZ5JTIyJTNBZmFsc2UlMkMlMjJzY29wZUhvaXN0JTIyJTNBZmFsc2UlMkMlMjJzb3VyY2VNYXBzJTIyJTNBdHJ1ZSUyQyUyMnB1YmxpY1VybCUyMiUzQSUyMiUyRl9fcmVwbF9kaXN0JTIyJTJDJTIydGFyZ2V0VHlwZSUyMiUzQSUyMmJyb3dzZXJzJTIyJTJDJTIydGFyZ2V0RW52JTIyJTNBbnVsbCUyQyUyMm91dHB1dEZvcm1hdCUyMiUzQW51bGwlMkMlMjJobXIlMjIlM0F0cnVlJTJDJTIybW9kZSUyMiUzQSUyMmRldmVsb3BtZW50JTIyJTJDJTIycmVuZGVyR3JhcGhzJTIyJTNBZmFsc2UlMkMlMjJ2aWV3U291cmNlbWFwcyUyMiUzQWZhbHNlJTJDJTIyZGVwZW5kZW5jaWVzJTIyJTNBJTVCJTVEJTJDJTIybnVtV29ya2VycyUyMiUzQTAlN0QlMkMlMjJ1c2VUYWJzJTIyJTNBZmFsc2UlMkMlMjJicm93c2VyQ29sbGFwc2VkJTIyJTNBJTVCJTVEJTJDJTIydmlld3MlMjIlM0ElNUIlMjIlMkZzcmMlMkZpbmRleC5odG1sJTIyJTJDJTIyJTJGc3JjJTJGaW5kZXguanN4JTIyJTJDJTIyJTJGc3JjJTJGQXBwLmpzeCUyMiUyQyUyMiUyRnBhY2thZ2UuanNvbiUyMiU1RCUyQyUyMmN1cnJlbnRWaWV3JTIyJTNBMyU3RA==).

While the solution appears to depend on one of the upstream libraries, these things that seem to fix the issue:

- Ditch the nullish coalescing operator
- Using an explicit `return` instead of implicit in the arrow operator

**Comment by snowystinger** [comment]
Thanks for providing the reproduction, we'll definitely want to get that fixed. I've opened the bug in Parcel as well https://github.com/parcel-bundler/parcel/issues/9947

In the meantime, patching such as patch-package should be able to help users work around the problem.

**Comment by felixwaldbach** [comment]
Hi, FYI I am having the same issue with the pure comments in this file:

https://www.unpkg.com/browse/react-aria-components@1.3.3/dist/Breadcrumbs.mjs

**Comment by snowystinger** [comment]
Looks like it's fixed in Parcel. https://github.com/parcel-bundler/parcel/issues/9947#event-14448749509
When we update this should be resolved.

**Comment by happenslol** [comment]
I'm still getting this warning with vite 5.4.11 and react-aria-components 1.4.1.

**Comment by snowystinger** [comment]
This should be fixed now. Code looks very different than the original report. We can reopen with a new reproduction if it's still an issue.
